### ISSUE TITLE
Why there is 'self.train' function in generate method for inference?

### ISSUE THREAD

**Issue by begeekmyfriend** [fatchord/WaveRNN#71]
I find that you have added [self.train](https://github.com/fatchord/WaveRNN/blob/master/models/fatchord_version.py#L241) method in `WaveRNN.generate` method. It works fine anyway. However when I commented out this line of code there seemed no difference for the evaluation. So I would like to know what is the use of `self.train` here?

**Comment by begeekmyfriend** [comment]
Hope I have got the point. Is it a better thing that if you put `self.train` in the line after [model.checkpoint](https://github.com/fatchord/WaveRNN/blob/master/train_wavernn.py#L57)?

**Comment by fatchord** [comment]
@begeekmyfriend It's a habit I have more than anything. Whenever I am generating from a model I will use `model.eval()`, do the generating code,  and then `model.train()`. It's in case there's stuff like dropout and so on in the model.
